lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -3383,48 +3384,47 @@ def test_permuations(self):
+                assert courseware_mfe_is_advertised(new_course_key, False, True)
 
                 # (Regular users would see the link, but they can't see the Legacy
                 #  experience, so it doesn't matter.)
 
             with _set_mfe_flag(False):
                 # (preview=on, redirect=off)
                 # Global and Course Staff can see the link.
-                assert show_courseware_mfe_link(global_staff_user, True, new_course_key)
-                assert show_courseware_mfe_link(global_staff_user, False, new_course_key)
-                assert show_courseware_mfe_link(regular_user, True, new_course_key)
+                assert courseware_mfe_is_advertised(new_course_key, True, True)
+                assert courseware_mfe_is_advertised(new_course_key, True, False)
+                assert courseware_mfe_is_advertised(new_course_key, False, True)
 
                 # Regular users don't see the link.
-                assert not show_courseware_mfe_link(regular_user, False, new_course_key)
+                assert not courseware_mfe_is_advertised(new_course_key, False, False)
 
         with _set_preview_mfe_flag(False):
             with _set_mfe_flag(True):
                 # (preview=off, redirect=on)
                 # Global staff see the link anyway
-                assert show_courseware_mfe_link(global_staff_user, True, new_course_key)
-                assert show_courseware_mfe_link(global_staff_user, False, new_course_key)
+                assert courseware_mfe_is_advertised(new_course_key, True, True)
+                assert courseware_mfe_is_advertised(new_course_key, True, False)
 
                 # If redirect is active for their students, course staff see the link even
                 # if preview=off.
-                assert show_courseware_mfe_link(regular_user, True, new_course_key)
+                assert courseware_mfe_is_advertised(new_course_key, False, True)
 
                 # (Regular users would see the link, but they can't see the Legacy
                 #  experience, so it doesn't matter.)
 
             with _set_mfe_flag(False):
                 # (preview=off, redirect=off)
-                # Global staff see the link anyway
-                assert show_courseware_mfe_link(global_staff_user, True, new_course_key)
-                assert show_courseware_mfe_link(global_staff_user, False, new_course_key)
-
-                # Course teams can NOT see the link because both rollout waffle flags are false.
-                assert not show_courseware_mfe_link(regular_user, True, new_course_key)
+                # Global staff  and course teams can NOT see the link
+                # because both rollout waffle flags are false.
+                assert not courseware_mfe_is_advertised(new_course_key, True, True)
+                assert not courseware_mfe_is_advertised(new_course_key, True, False)
+                assert not courseware_mfe_is_advertised(new_course_key, False, True)
```

Comment:
These are the assertions that actually changed. Namely, global staff can longer see the link "View in New Experience" link if the preview is off and the MFE is inactive.